npm/pkg/dataplane/dataplane.go

```diff
@@ -236,9 +268,18 @@ func (dp *DataPlane) ApplyDataPlane() error {
 		// lock updatePodCache while driving goal state to kernel
 		// prevents another ApplyDataplane call from updating the same pods
 		dp.updatePodCache.Lock()
-		defer dp.updatePodCache.Unlock()
+		defer func() {
+			dp.updatePodCache.removeDeletedItemsFromQueue()
+			dp.updatePodCache.Unlock()
```

Comment:
May be create a datastructure called ordered Map ? and use that as the update pod cache ? 

example: https://github.com/iancoleman/orderedmap/blob/master/orderedmap.go